### ISSUE TITLE
Changing the weight of Nimbus Mono freezes FontForge

### ISSUE THREAD

**Issue by kirbykevinson** [fontforge/fontforge#5036]
* Fedora 36, both RPM and Flatpak
* FontForge 20220308

![Screenshot from 2022-06-05 12-37-15](https://user-images.githubusercontent.com/17949857/172036850-d9362e54-54d8-45f8-947f-3d4c7dcb8dd7.png)

![Screenshot from 2022-06-05 12-22-53](https://user-images.githubusercontent.com/17949857/172036448-bf59b648-14c7-4c1e-833f-30cc56316756.png)

When you try to change the weight of the Nimbus Mono font to 25 (compared to default 50), FontForge freezes when reaching either the pipe or the section sign character. In order to reproduce the bug, [this version](https://github.com/ArtifexSoftware/urw-base35-fonts/blob/master/fonts/NimbusMonoPS-Regular.otf) of the font was used. 

While freezing, the following logs are produced:

```
Internal Error (overlap): Mismatched intersection.
 (540.5,491)->(517.955,458.009) ends at (-999999,-999999) while (517.955,458.009)->(171.955,270.009) starts at (517.955,458.009)
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): monotonic is both needed and unneeded (540.5,491)->(517.955,458.009). x=517.955 (prev=507.791)
Internal Error (overlap): monotonic is both needed and unneeded (502,531.5)->(533.608,512.612). x=517.955 (prev=335.868)
Internal Error (overlap): Winding number did not return to 0 when x=517.955
Internal Error (overlap): Expected needed monotonic @(517.955,458.009) (517.955,458.009)->(517.955,458.009).
Internal Error (overlap): Expected needed monotonic @(517.955,458.009) (517.955,458.009)->(517.955,458.009).
```

**Comment by skef** [comment]
This could be a dup of #5012. Submitter is encouraged to try to reproduce with code from PR #5013 if they can.

**Comment by kirbykevinson** [comment]
Yup, #5013 fixes it.

**Comment by ctrlcctrlv** [comment]
#5013 hasn't been merged so the close by @kirbykevinson was premature; reopening.

**Comment by skef** [comment]
@ctrlcctrlv it was closed as a dup of #5012. I don't see why we need it open but if you want it that way you should add it to the #5013 close list.
